building/configuring_git_workflow.adoc

```diff
@@ -0,0 +1,426 @@
+= Configuring Git Workflow
+
+== How to setup git locally
+
+The process for setting up git varies based on your operating system. See http://git-scm.com/book/en/v2/Getting-Started-Installing-Git[Getting Started Installing Git] for more information. Once git is installed, continue with the steps below. 
+
+=== Setting up your ~/.gitconfig 
+
+In order to have patches accepted, you should first declare your identity by 
+modifying your `~/.gitconfig` file to add the following:
+
+[source,bash]
+----
+[user]
+        name = Your Name
+        email = yourname@example.com
+----
+
+==== Useful aliases for the ~/.gitconfig
+
+Here are some aliases you can add to your `~/.gitconfig` file, in the `[alias]` section. 
+
+The first two aliases (`lg` and `lg1`) assist you in browsing the commit log. 
+After adding them to your .gitconfig file, you can test them out by typing `git lg 5`, 
+which means show the past 5 commits in a nice pretty format. 
+
+The third alias, `ch`, allows you to rebase the last `n` commits, and to either 
+modify the commit messages, squash the entries, re-order entries, and other functions native to git. 
+
+[source,bash]
+----
+[alias]
+  lg = log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %C(blue)%aE%Creset %Cgreen(%cr)%Creset' --abbrev-commit --date=relative
+  lgl = log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s \n\t\t\t%C(blue)%aN %aE, %cN %cE%Creset %Cgreen(%cD - %cr)%Creset' --abbrev-commit --date=relative
+  ch = "!sh -c 'git rebase -i HEAD~$1' -"
```

Comment:
"ch" is also short for "change". I use it for more than just squashing -- you can rebase the last `n` commits, modify commit messages, squash (combine) commits or re-order commits. Thus to me "change" makes more sense than "squash". I've reworked the copy a little however:

==== Useful aliases for the ~/.gitconfig

Here are some aliases you can optionally add to your `~/.gitconfig` file, in the `[alias]` section. 

The first alias, `lg`, assists you in browsing the commit log. 
After adding it to your .gitconfig file, try `git lg 5` to show the last 5 commits. 

The second alias, `ch`, allows you to rebase the last `n` commits, modify commit messages, squash (combine) commits or re-order commits. 
Try `git ch 3` to work with your last 3 commits.

[source,bash]
----
[alias]
  lg = log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %C(blue)%aE%Creset %Cgreen(%cr)%Creset' --abbrev-commit --date=relative
  ch = "!sh -c 'git rebase -i HEAD~$1' -"
----